addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -0,0 +1,948 @@
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.tesla.internal.protocol.ChargeState;
+import org.openhab.binding.tesla.internal.protocol.ClimateState;
+import org.openhab.binding.tesla.internal.protocol.DriveState;
+import org.openhab.binding.tesla.internal.protocol.GUIState;
+import org.openhab.binding.tesla.internal.protocol.TokenRequest;
+import org.openhab.binding.tesla.internal.protocol.Vehicle;
+import org.openhab.binding.tesla.internal.protocol.VehicleState;
+import org.openhab.binding.tesla.internal.throttler.QueueChannelThrottler;
+import org.openhab.binding.tesla.internal.throttler.Rate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link TeslaHandler} is responsible for handling commands, which are sent
+ * to one of the channels.
+ * 
+ * @author Karel Goderis - Initial contribution
+ */
+public class TeslaHandler extends BaseThingHandler {
+
+	public static final int EVENT_REFRESH_INTERVAL = 200;
+	public static final int FAST_STATUS_REFRESH_INTERVAL = 15000;
+	public static final int SLOW_STATUS_REFRESH_INTERVAL = 60000;
+	public static final int MINIMUM_EVENT_INTERVAL = 15000;
+
+	private Logger logger = LoggerFactory.getLogger(TeslaHandler.class);
+
+	// Vehicle state variables
+	protected Vehicle vehicle;
+	protected String vehicleJSON;
+	protected DriveState driveState;
+	protected GUIState guiState;
+	protected VehicleState vehicleState;
+	protected ChargeState chargeState;
+	protected ClimateState climateState;
+	protected String accessToken;
+
+	// REST Client API variables
+	protected Client teslaClient = ClientBuilder.newClient();
+	protected Client eventClient;
+	protected WebTarget teslaTarget = teslaClient.target(TESLA_OWNERS_URI);
+	protected WebTarget tokenTarget = teslaTarget.path(TESLA_ACCESS_TOKEN_URI);
+	protected WebTarget vehiclesTarget = teslaTarget.path(API_VERSION).path(
+			VEHICLES);
+	protected WebTarget vehicleTarget = vehiclesTarget.path(VEHICLE_ID_PATH);
+	protected WebTarget dataRequestTarget = vehicleTarget
+			.path(DATA_REQUEST_PATH);
+	protected WebTarget commandTarget = vehicleTarget.path(COMMAND_PATH);
+	protected WebTarget eventTarget;
+
+	// Threading and Job related variables
+	protected final ScheduledThreadPoolExecutor scheduler = (ScheduledThreadPoolExecutor) Executors
```

Comment:
Yes, I do not want to block other bindings that rely on that pool, and I  have encapsulated each request to the Car as a Runnable. Ok, they are throttled, but when I wrote the code I opted to give the binding its own pool in order to prevent any side-events. I though the existing pool was a bit limited in size as well